File: hw3/city.js
class City {
    constructor(size, name, population) {
        this._size = size;
        this._name = name;
        this._population = population;
    }

    isNewCity() {
        return this._population == 0;
    }

    density() {
        if(this._population < 100000) {
            return "Town";
        } else if(this._population < 10000000) {
            return "City";
        } else {
            return "MegaCity";
        }
    }

    get population() {
        return this._population;
    }

    get size() {
        return this._size;
    }

    get name() {
        return this._name;
    }

    set landSize(area) {
    if(area <= 0) {
        throw Error('Error, Not Enough Land!');
    }
    this._size = area;
}
}

module.exports = {'City' : City }

File: hw4/city.py
from typing import List

class Neighborhood:
    def __init__(self, sqMiles : float, education : str) :
        self._sqMiles = sqMiles
        self._education = education

    @property
    def sqMiles(self):
        return self._sqMiles

    @property
    def education(self):
        return self._education

class Houses:
    pass

class City(Neighborhood):
    def __init__(self, sqMiles : float, education : str, size : float, name : str, population : int = 0) -> None:
        Neighborhood.__init__(self, sqMiles, education)
        self._size : float = size
        self._name : str = name
        self._population : int = population
        self._houses : List[Houses] = []

    def isNewCity(self) -> bool:
        return self.population == 0

    def density(self) -> str:
        if self._population < 100000:
            return "Town"
        elif self._population < 10000000:
            return "City"
        else: 
            return "MegaCity"

    @property
    def population(self):
        return self._population

    @property 
    def size(self):
        return self._size

    @property
    def name(self) -> str:
        return self._name

    @size.setter
    def landSize(self, area: float) -> None:
        if area < 0:
            raise ValueError('Error, Not Enough Land!')
        self._size = area
